rfcs/0134-nix-store-layer.md

```diff
@@ -0,0 +1,271 @@
+
+## 2. Manual
+
+It should be possible to build a store-only manual without information on the other layers too.
+This would be the manual that is distributed with the store-only Nix.
+Of course, store-only and full  can share sections, so we aren't duplicating work.
+
+## 3. Website
+
+The fully and store-only version of Nix should both be presented for download on the website.
+This should be just like how Plasma, Gnome, and headless installer images for NixOS are all offered.
+
+## 4. Store-specific Tests
+
+The current test suite uses Nix language for most tests of store-layer functionality.
+We should write new tests that don't use the Nix language.
+E.g. we might create a `read-derivation` complement of `show-derivation` that accepts a nicer JSON representation of a derivation as input.
+This will allow the store-only Nix to be tested in isolation, but these tests can also be used with full Nix.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+## Maximal Nix unlike today
+
+I cannot emphasize this enough, but the interface of maximal store+exprs+flakes Nix remains *exactly* like today.
+In particular, lower level commands can be used with higher level "installables" (arguments), so e.g.
+```
+nix show-derivation flake#bar
+```
+will still work.
+
+## Good tasks anyways!
+
+Lots of the plan above I think is good work we should be doing anyways, regardless of whether we expose a store-only Nix.
+If you believe this, then the "cost" of this RFC is a lot less.
+
+### Tests
+
+Splitting the test suite per natural layer of the implementation is good work because it combines the specificity of unit tests with the real-world-ness of integration tests.
+"entire kitchen sink" tests make it harder to narrow down root causes of failures.
+
+### Manual
+
+The documentation team is already working to clean up the manual, and this effort already involves emphasizing layering.
+So whether we formally make a store-only Nix or not, I suspect the overhauled manual will natural have easy boundaries from which to "extract" the store-only manual.
+
+## Security
+
+The daemon is a privileged process.
+Even if with upcoming changes it shouldn't need root, it does tasks like administrating OS sandboxes correctly which still are security-critical.
+Having less code in the story-only Nix daemon, even if we think the removed code was "dead anyways" is always good.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Creating new teams, trying to build ties with other communities sounds scary.
+
+# Alternatives
+[alternatives]: #alternatives
+
+Do programming parts without the documentation and website parts.
+But that feels to me like turning an ongoing shift in focus to a one-off change that is likely to bit-rot.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+What should the store-only Nix be called?
```

Comment:
`nix-store`

Reply:
I don't think we _need_ to decide on a new name. It could be just a build configuration of the already named components. 
For an analogy, in Nixpkgs both `docker` and `docker-client` packages provide the `docker` command, both of which are produced by the same package function.

On the other hand, renaming would be a nice gesture, if we drop `nix` from the names: `libstore`, `store add-file FILE`, etc.

Reply:
I've been liking:
- store: only the store layer
- nix: includes nix language
- flake: include flakes concepts

Reply:
I agree with @roberth's idea of not needing a rename. For the purposes of somebody wanting to make sure they only depend on the store-only Nix, we can have `nix --version` indicate that it's a store-only Nix.

Reply:
Moved to future work in 3194ad43a5c302b5b261eafb25c0f1426203dbee. If everyone is good with that we can mark this thread resolved.